tests/v2/validation/fleet/provisioning/new_cluster_existing_gitrepo_test.go

```diff
@@ -0,0 +1,301 @@
+//go:build validation
+
+package provisioning
+
+import (
+	"testing"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/v2/actions/clusters"
+	"github.com/rancher/rancher/tests/v2/actions/fleet"
+	"github.com/rancher/rancher/tests/v2/actions/provisioning"
+	"github.com/rancher/rancher/tests/v2/actions/provisioning/permutations"
+	"github.com/rancher/rancher/tests/v2/actions/provisioninginput"
+	"github.com/rancher/rancher/tests/v2/actions/reports"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/clusters/kubernetesversions"
+	extensionsfleet "github.com/rancher/shepherd/extensions/fleet"
+	"github.com/rancher/shepherd/extensions/users"
+	password "github.com/rancher/shepherd/extensions/users/passwordgenerator"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/environmentflag"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FleetWithProvisioningTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	provisioningConfig *provisioninginput.Config
+	fleetGitRepo       *v1alpha1.GitRepo
+}
+
+func (f *FleetWithProvisioningTestSuite) TearDownSuite() {
+	f.session.Cleanup()
+}
+
+func (f *FleetWithProvisioningTestSuite) SetupSuite() {
+	f.session = session.NewSession()
+
+	client, err := rancher.NewClient("", f.session)
+	require.NoError(f.T(), err)
+
+	f.client = client
+
+	// move this out of setup into a helper function / action
+	f.fleetGitRepo = &v1alpha1.GitRepo{
```

Comment:
Is this going to get moved into a helper function?

Reply:
I'll double check with @igomez06 as in the past, making a function call just to create the local object for you was frowned upon

Reply:
Honestly we don't need to abstract everything into functions. Since Caleb is just setting up an object I honestly think this is more readable than having it be its own func. Too much abstraction leads to difficulties in failure analysis.